apiserver/facades/client/modelgeneration/interface.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelgeneration
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/model"
+)
+
+// ModelGeneration defines the methods exported by the model generation API facade.
+type ModelGeneration interface {
+	AddGeneration() (params.ErrorResult, error)
+	AdvanceGeneration(args params.Entities) (params.ErrorResults, error)
+	SwitchGeneration(arg params.GenerationVersionArg) (params.ErrorResult, error)
+}
+
+// ModelGenerationState represents the state of an model required by the ModelGeneration.
```

Comment:
You could have one generation statement with all of the desired interfaces going into a single file, but it's all the same.